### PR TITLE
transparently control apache2_mod_ properties via apache2_module

### PR DESCRIPTION
**Pull request by bottkv488**
# Description

* Add property Hash `:mod_conf`to `apache2_module`
* validate keys as existing properties
* send all key, value pairs as properties to `apache2_mod_#{new_resource.name}` 
* provide basic error handling for failing resource lookup

## Issues Resolved

https://github.com/sous-chefs/apache2/issues/600

## Check List

- [ ] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: resources/module.rb [hunk 1]
```diff
@@ -23,7 +27,27 @@
...
+  if new_resource.conf
+    mod_resource_name = "apache2_mod_#{new_resource.name}"
+
+    send(mod_resource_name, 'default')
+
+    # pass Hash new_resource.mod_conf as properties
+    begin
+      r = resources(mod_resource_name.to_sym => 'default')
+
+      new_resource.mod_conf.each do |k, v|
```
Comment (damacus) [inline resources/module.rb]: I would prefer better names here, so the next person can decipher what we did :)

File: resources/mod_proxy.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
          description: ''
 
 property :require, String,
-         default: '',
+         default: 'all denied',
          description: ''
```
Comment (damacus) [inline resources/mod_proxy.rb]: We could do with a description here. And then copy that description to the documentation (which doesn't exist yet 😭 )

File: resources/module.rb [hunk 3]
```diff
@@ -12,6 +12,10 @@
          default: lazy { "#{name}_module" },
          description: 'String to identify the module for the `LoadModule` directive'
 
+property :mod_conf, Hash,
+        default: {},
+        description: 'Pass properties to apache2_mod_<name>'
```
Comment (damacus) [inline resources/module.rb]: Throw this in the docs, and I think this is good 👍

File: resources/module.rb [hunk 4]
```diff
@@ -23,7 +27,27 @@
 
 action :enable do
   # Call the  apache2_mod_resource if we want it configured
-  send("apache2_mod_#{new_resource.name}", '') if new_resource.conf
+  if new_resource.conf
+    mod_resource_name = "apache2_mod_#{new_resource.name}"
```
Comment (damacus) [inline resources/module.rb]:
```suggestion
    mod_resource_name = "apache2_mod_#{new_resource.name}".to_sym
```

File: resources/module.rb [hunk 5]
```diff
@@ -23,7 +27,27 @@
...
+
+    send(mod_resource_name, 'default')
+
+    # pass Hash new_resource.mod_conf as properties
+    begin
+      r = resources(mod_resource_name.to_sym => 'default')
```
Comment (damacus) [inline resources/module.rb]:
```suggestion
      r = resources(mod_resource_name=> 'default')
```

**Comment by damacus** [comment]
🎉

**Pull request merged**